src/Stratis.Features.FederatedPeg/SourceChain/MaturedBlocksProvider.cs

```diff
@@ -41,10 +43,15 @@ public MaturedBlocksProvider(ILoggerFactory loggerFactory, IDepositExtractor dep
         }
 
         /// <inheritdoc />
-        public async Task<Result<List<MaturedBlockDepositsModel>>> GetMaturedDepositsAsync(int blockHeight, int maxBlocks)
+        public async Task<Result<List<MaturedBlockDepositsModel>>> GetMaturedDepositsAsync(int blockHeight, int maxBlocks, int maxDeposits = int.MaxValue)
         {
             ChainedHeader consensusTip = this.consensusManager.Tip;
 
+            if (consensusTip == null)
+            {
+                return Result<List<MaturedBlockDepositsModel>>.Fail("Not ready to provide blocks.");
```

Comment:
Change to `ConsensusManager's tip has not yet been initialized.`

Reply:
That message would make sense to us but not to a user. The message was chosen in this way to be user-friendly.

Reply:
This message gets displayed on the console, so it might be of worth to the user to know why the other node isnt ready yet?